docs/en/building/supplemental.md

```diff
@@ -0,0 +1,541 @@
+# Creating Bundles
+
+A bundle is a portion of code that represents a slice of functionality. Bundles can be loaded asynchronously on demand and in parallel. An application that is appropriately bundled can be significantly more responsive and require fewer kilobytes and less load time than an application that does not use any kind of code splitting. This is especially important when working with large applications where much of the presentation logic isn't needed on the initial load.
+
+Dojo tries to make intelligent choices by using routes and outlets to automatically split code into smaller bundles. In general these bundles should all have code that is related and relevant. This comes for free as part of the build system and doesn't require any additional thought to use. However, for those with specific bundling needs Dojo also allows for bundles to be explicitly defined in the `.dojorc` configration file.
+
+By default a Dojo application only creates a single application bundle. However, there are a number of configuration options provided by [@dojo/cli-build-app](https://github.com/dojo/cli-build-app) that will help break down an application into smaller portions that can be progressively loaded.
+
+## Automatic bundling using routes
+
+By default Dojo will create bundles based on the routes used by your application. In order to do this several rules must be followed.
+
+1.  `src/routes.ts` must have a default export containing the routing configuration
+1.  Widgets must be the default export of their module
+1.  `Outlet`s `render` function must use inline functions
+
+> src/routes.ts
+
+```ts
+export default [
+	{
+		path: 'home',
+		outlet: 'home',
+		defaultRoute: true
+	},
+	{
+		path: 'about',
+		outlet: 'about'
+	},
+	{
+		path: 'profile',
+		outlet: 'profile'
+	}
+];
+```
+
+> src/App.ts
+
+```ts
+export default class App extends WidgetBase {
+	protected render() {
+		return v('div', { classes: [css.root] }, [
```

Comment:
These new ref guides are effectively launching as v6's documentation, so preferring the latest & greatest authoring styles when giving widget examples. This is mainly:

* Preferring `tsx` output over `v()`/`w()`
* Moving to functional widgets (we'll need to do a pass over all guides once this feature has stabilized, so we can keep examples as class-based for the moment)

Reply:
Do we have a style guide somewhere that lists these preferences? This code was pulled directly from `@dojo/cli-create-app` and was used for consistency from the basics guide.